### PR TITLE
python3Packages.pyftdi: 0.51.2 -> 0.52.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 0.52.0

Disable the tests as they require access to the serial port and fail.

```bash
======================================================================
ERROR: protocol_unix (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: protocol_unix
Traceback (most recent call last):
  File "/nix/store/m7080pw0ryjk0jhljp55rq1hd2qy8gki-python3-3.8.6/lib/python3.8/unittest/loader.py", line 154, in loadTestsFromName
    module = __import__(module_name)
  File "/build/source/pyftdi/serialext/protocol_unix.py", line 32, in <module>
    from serial import (SerialBase, SerialException, portNotOpenError,
ImportError: cannot import name 'portNotOpenError' from 'serial' (/nix/store/ivmwqcnbzarm1gh430hwhbwy3bpb34yy-python3.8-pyserial-3.5/lib/python3.8/site-packages/serial/__init__.py)
```

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 107982` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python37Packages.alarmdecoder</li>
    <li>python37Packages.pyftdi</li>
    <li>python38Packages.alarmdecoder</li>
    <li>python38Packages.pyftdi</li>
    <li>python39Packages.alarmdecoder</li>
    <li>python39Packages.pyftdi</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyftdi/default.nix [hunk 1]
```diff
@@ -1,21 +1,30 @@
...
-    sha256 = "14mkwk44bgm6s4kqagz7nm6p6gsygmksl2628jaqh7ppblxca9as";
+    sha256 = "0nm4z7v9qcb9mxqbl21jgzica4faldnpy5qmbkrc6scnx55pxfm9";
   };
 
   propagatedBuildInputs = [ pyusb pyserial ];
 
   pythonImportsCheck = [ "pyftdi" ];
 
+  # tests requires access to the serial port
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyftdi/default.nix:26]:
```suggestion
  # tests requires access to the serial port
  doCheck = false;
  pythonImportsCheck = [ "pyftdi" ];
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107982` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python37Packages.alarmdecoder</li>
    <li>python37Packages.pyftdi</li>
    <li>python38Packages.alarmdecoder</li>
    <li>python38Packages.pyftdi</li>
    <li>python39Packages.alarmdecoder</li>
    <li>python39Packages.pyftdi</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107982` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python37Packages.alarmdecoder</li>
    <li>python37Packages.pyftdi</li>
    <li>python38Packages.alarmdecoder</li>
    <li>python38Packages.pyftdi</li>
    <li>python39Packages.alarmdecoder</li>
    <li>python39Packages.pyftdi</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 107982` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python37Packages.alarmdecoder</li>
    <li>python37Packages.pyftdi</li>
    <li>python38Packages.alarmdecoder</li>
    <li>python38Packages.pyftdi</li>
    <li>python39Packages.alarmdecoder</li>
    <li>python39Packages.pyftdi</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

File: pkgs/development/python-modules/pyftdi/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/pyftdi/default.nix]: Updated. Thanks

File: pkgs/development/python-modules/pyftdi/default.nix [hunk 2]
```diff
@@ -1,24 +1,39 @@
...
   meta = {
     description = "User-space driver for modern FTDI devices";
+    longDescription = ''
+      PyFtdi aims at providing a user-space driver for popular FTDI devices.
+      This includes UART, GPIO and multi-serial protocols (SPI, I2C, JTAG)
+      bridges.
+    '';
     homepage = "https://github.com/eblot/pyftdi";
     license = lib.licenses.bsd3;
+    maintainers = with maintainers; [ fab ];
```
Comment (jonringer) [inline pkgs/development/python-modules/pyftdi/default.nix:37]:
```suggestion
    maintainers = with lib.maintainers; [ fab ];
```

```
at: (37:24) in file: /home/jon/.cache/nixpkgs-review/pr-107982/nixpkgs/pkgs/development/python-modules/pyftdi/default.nix

    36|     license = lib.licenses.bsd3;
    37|     maintainers = with maintainers; [ fab ];
      |                        ^
    38|   };

undefined variable 'maintainers'
```

**Review by jonringer — COMMENTED** [review]
or do `meta = with lib; {`

File: pkgs/development/python-modules/pyftdi/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/development/python-modules/pyftdi/default.nix:37]: Thanks.

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 107982` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python37Packages.alarmdecoder</li>
    <li>python37Packages.pyftdi</li>
    <li>python38Packages.alarmdecoder</li>
    <li>python38Packages.pyftdi</li>
    <li>python39Packages.alarmdecoder</li>
    <li>python39Packages.pyftdi</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107982` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python37Packages.alarmdecoder</li>
    <li>python37Packages.pyftdi</li>
    <li>python38Packages.alarmdecoder</li>
    <li>python38Packages.pyftdi</li>
    <li>python39Packages.alarmdecoder</li>
    <li>python39Packages.pyftdi</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review. If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107982` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>python37Packages.alarmdecoder</li>
<li>python37Packages.pyftdi</li>
<li>python38Packages.alarmdecoder</li>
<li>python38Packages.pyftdi</li>
<li>python39Packages.alarmdecoder</li>
<li>python39Packages.pyftdi</li>
</ul>
</details>

**Pull request merged**